src/main/java/org/opensearch/ad/ratelimit/CheckpointWriteWorker.java

```diff
@@ -188,11 +188,15 @@ public void write(ModelState<EntityModel> modelState, boolean forceWrite, Reques
                     return;
                 }
 
+                modelState.setLastCheckpointTime(clock.instant());
                 CheckpointWriteRequest request = new CheckpointWriteRequest(
                     System.currentTimeMillis() + detector.getDetectorIntervalInMilliseconds(),
                     detectorId,
                     priority,
-                    new IndexRequest(indexName).id(modelId).source(source)
+                    // If the document does not already exist, the contents of the upsert element
+                    // are inserted as a new document.
+                    // ßIf the document exists, update fields in the map
```

Comment:
nit: Is first character in this comment intentional?

Reply:
ah, removed